File: backend/src/main/java/org/example/backend/recipe/model/RecipeDetailModel.java
package org.example.backend.recipe.model;

import com.mongodb.lang.Nullable;

import java.time.LocalDateTime;
import java.util.List;

public record RecipeDetailModel(
        String _id,
        String title,
        List<IngredientGroupModel> ingredients,
        List<StepGroupModel> steps,
        NutritionValuesModel nutritionValues,
        LocalDateTime createdAt,
        LocalDateTime updatedAt,
        int preparationTime,
        int cookingTime,
        int totalTime,
        int defaultPortions,
        @Nullable List<String> allergens,
        List<String> tags,
        String image
) {
}

File: frontend/src/models/recipe/RecipeDetailModel.ts
import type { IngredientGroupModel } from "./IngredientGroupModel.ts";
import type { NutritionValuesModel } from "./NutritionValuesModel.ts";
import type { StepGroupModel } from "./StepGroupModel.ts";

export type RecipeDetailModel = {
  id: string;
  title: string;
  ingredients: IngredientGroupModel[];
  steps: StepGroupModel[];
  nutritionValues: NutritionValuesModel;
  createdAt: string;
  updatedAt: string;
  preparationTime: number;
  cookingTime: number;
  totalTime: number;
  defaultPortions: number;
  allergens: string[] | null;
  tags: string[];
  image: string;
};
